src/Stryker.Core/Stryker.Core/Mutants/MutationContext.cs

```diff
@@ -47,10 +56,28 @@ public MutationContext Clone()
             return new MutationContext(this);
         }
 
+        public void Discard()
+        {
+            if (HasStatementLevelMutant)
+            {
+                // some mutants 
+                Logger.LogDebug($"{BlockLevelControlledMutations.Count+StatementLevelControlledMutations.Count} mutations were not injected.");
+                foreach (var mutant in BlockLevelControlledMutations.Union(StatementLevelControlledMutations))
+                {
+                    mutant.ResultStatus = MutantStatus.CompileError;
+                    mutant.ResultStatusReason = "Stryker was not able to inject mutation in code.";
```

Comment:
This does not sound like a compile error. Do we need a new class of error for this? Or should these mutations perhaps simply be removed completely instead of marked as error?

Reply:
We can't remove mutations as this will leave unused indices and will look suspicious.
I guess this could be part of a broader discussion, for example, I am not sure of the value of the 'CompileError' status in general.

Reply:
Agreed